### PR TITLE
Tweak logs and CLI

### PR DESCRIPTION
**Pull request by ouziel-slama**
Capture d’écran 2024-05-27 à 16 24 42

in the log file:
```
{"filename": "server.py", "funcName": "start_all", "levelname": "CRITICAL", "lineno": 670, "module": "server", "name": "counterparty", "pathname": "/home/ouziel/counterparty/counterparty-core/counterparty-core/counterpartycore/server.py", "process": 263922, "processName": "MainProcess", "stack_info": null, "thread": 139909791457280, "threadName": "MainThread", "message": "test critical color", "time": "2024-05-27T14:23:09.981966+00:00"}
{"event": {"name": "ORDER_UPDATE", "block_index": 30000, "tx_hash": "1234567", "status": "success"}, "filename": "log.py", "funcName": "log_event", "levelname": "INFO", "lineno": 159, "module": "log", "name": "counterparty", "pathname": "/home/ouziel/counterparty/counterparty-core/counterparty-core/counterpartycore/lib/log.py", "process": 263922, "processName": "MainProcess", "stack_info": null, "thread": 139909791457280, "threadName": "MainThread", "message": "Block 30000 - Updated order 1234567. New status: success", "time": "2024-05-27T14:23:09.982216+00:00"}
{"event": {"name": "ORDER_MATCH_EXPIRATION", "block_index": 30000, "order_match_id": "1234567_1234567"}, "filename": "log.py", "funcName": "log_event", "levelname": "INFO", "lineno": 159, "module": "log", "name": "counterparty", "pathname": "/home/ouziel/counterparty/counterparty-core/counterparty-core/counterpartycore/lib/log.py", "process": 263922, "processName": "MainProcess", "stack_info": null, "thread": 139909791457280, "threadName": "MainThread", "message": "Block 30000 - Order match 1234567_1234567 expired", "time": "2024-05-27T14:23:09.982505+00:00"}
```

### CODE REVIEW

**Comment by ouziel-slama** [comment]
OK so there are two types of output for the CLI:

1. with pretty spinners and colorful `print()`
2. colorized speaking-english-human-readable logs

- [x] The first is used by default for all commands except `start`.
- [x] The second is always used with the `--verbose` flag.
- [x] Logs are saved in the `server.log` file in Json format
- [x] Hashes and addresses are truncated
- [x] each log level is displayed in a different color
- [x] a new log level is introduced: `EVENT`
- [x] when possible all messages are prefixed with `Block N` or `Mempool`
- [ ] prefixed API request logs with `API Request`
- [ ] Log full requests, including POST payload, when --verbose is set.

**Comment by adamkrellenstein** [comment]
Okay, a couple of comments:

1. We don't want the hashes / addresses truncated in the JSON logs.
2. We don't want non-block messages like `PRAGMA optimize done` to have the block prefix
3. Can we have this appendix for transactions?
Screenshot 2024-05-28 at 5 10 02 PM
4. The last message for a block should be `INFO] - Block 500000 - L: 114c052, TX: 78a2b9a, M: d2e003b (2.0s)`
5. The first block log entry should just be `2024-03-09-T13:08:10Z - [    INFO] - *Block 833899*`

**Comment by ouziel-slama** [comment]
- [x] We don't want the hashes / addresses truncated in the JSON logs.
- [x] We don't want non-block messages like PRAGMA optimize done to have the block prefix
- [ ] The last message for a block should be INFO] - Block 500000 - L: 114c052, TX: 78a2b9a, M: d2e003b (2.0s)
- [x] The first block log entry should just be 2024-03-09-T13:08:10Z - [    INFO] - *Block 833899*

**Comment by adamkrellenstein** [comment]
Can we also change the formatting for transactions to look like this?
Screenshot 2024-05-29 at 9 41 13 AM

Is the CLI output sent to `STDIN` and the log-like messages sent to `STDERR`?

**Comment by ouziel-slama** [comment]
> Can we also change the formatting for transactions to look like this? Screenshot 2024-05-29 at 9 41 13 AM
> 

done here https://github.com/CounterpartyXCP/counterparty-core/pull/1827/commits/32c30fa3dd133129ec2e7c89ac766ad49107d333

> Is the CLI output sent to `STDIN` and the log-like messages sent to `STDERR`?

yep! `print()` on STDOUT and `logger.*()` on STDERR.

File: counterparty-core/counterpartycore/lib/blocks.py [hunk 1]
```diff
@@ -1068,6 +1059,7 @@ def parse_new_block(db, decoded_block, block_parser=None, tx_index=None):
     decoded_block["block_index"] = util.CURRENT_BLOCK_INDEX
 
     with db:  # ensure all the block or nothing
+        logger.info(colored(f"Block {decoded_block['block_index']}", attrs=["bold"]))
```
Comment (adamkrellenstein) [inline counterparty-core/counterpartycore/lib/blocks.py:1062]: okay I changed my mind! I think the whole line including the timestamp and the log level should be bold here.

**Review by adamkrellenstein — APPROVED** [review]

File: counterparty-core/counterpartycore/lib/blocks.py (hunk 1 above)
Comment (ouziel-slama) [reply counterparty-core/counterpartycore/lib/blocks.py]: :) done!

**Pull request merged**